src/pytorch_lightning/strategies/sharded.py

```diff
@@ -120,14 +125,14 @@ def _reinit_optimizers_with_oss(self, optimizers: List[Union[Optimizer, Lightnin
                 del optimizer
         return optimizers
 
-    def optimizer_state(self, optimizer: "OSS") -> Optional[dict]:
+    def optimizer_state(self, optimizer: Optimizer) -> Dict[str, Any]:
```

Comment:
wondering, how it could not be typed as `LightningOptimizer` since there's a check inside
```py
if isinstance(optimizer, LightningOptimizer)
```

Reply:
Thank you @rohitgr7 for review. i'm not sure too, but mypy shows error, `optimizer_state` in strategy.py has `optimizer: Optimizer`
```
src/pytorch_lightning/strategies/sharded.py:129: error: Argument 1 of "optimizer_state" is incompatible with supertype "Strategy"; supertype defines the argument type as "Optimizer"  [override]
src/pytorch_lightning/strategies/sharded.py:129: note: This violates the Liskov substitution principle
src/pytorch_lightning/strategies/sharded.py:129: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#incompatible-overrides
src/pytorch_lightning/strategies/sharded.py:131: error: Incompatible types in assignment (expression has type "Optimizer", variable has type "LightningOptimizer")  [assignment]
```
so how about changing like this?
```suggestion
    def optimizer_state(self, optimizer: Union[Optimizer, LightningOptimizer]) -> Dict[str, Any]:
```

Reply:
Hey 
It is because LightningOptimizer does not inherit from Optimizer. It is imitating the full Optimizer by duck typing, but for mypy this is not visible. So I think we need to list it. 

@rohitgr7 If you find a better way, would you mind adding your suggestion directly to the branch so we can get this merged :)

Reply:
I guess this can be resolved, since `def optimizer_state` is not longer here due to #14208 being merged